modules/tinymce/src/core/main/ts/delete/InlineBoundaryDelete.ts

```diff
@@ -89,42 +87,49 @@ const backspaceDeleteCollapsed = (editor: Editor, caret: Cell<Text>, forward: bo
         Fun.constant(Optional.some(BoundaryLocation.inside(location)))  // After
       );
     }
-  })
-    .map(setCaretLocation(editor, caret))
-    .getOrThunk(() => {
-      const toPosition = CaretFinder.navigate(forward, rootNode, from);
-      const toLocation = toPosition.bind((pos) => BoundaryLocation.readLocation(isInlineTarget, rootNode, pos));
-
-      return Optionals.lift2(fromLocation, toLocation, () =>
-        InlineUtils.findRootInline(isInlineTarget, rootNode, from).exists((elm) => {
-          if (hasOnlyTwoOrLessPositionsLeft(elm)) {
+  });
+
+  const newLocation = location.map(setCaretLocation(editor, caret));
```

Comment:
This was mostly for trying to figure out what was going on, since the chaining made it confusing to read in some spots, and didn't show types, but it's not needed now so I'll chain it back up.